api/src/paths/access-request.ts

```diff
@@ -0,0 +1,160 @@
+import { addSystemUser } from './user';
+import { addSystemRoles } from './user/{userId}/system-roles';
+
+const defaultLog = getLogger('paths/access-request');
+
+export const PUT: Operation = [logRequest('paths/access-request', 'POST'), updateAccessRequest()];
+
+PUT.apiDoc = {
+  description: "Update a user's system access request and add any specified system roles to the user.",
+  tags: ['user'],
+  security: [
+    {
+      Bearer: WRITE_ROLES
+    }
+  ],
+  requestBody: {
+    content: {
+      'application/json': {
+        schema: {
+          type: 'object',
+          required: ['userIdentifier', 'identitySource', 'requestId', 'requestStatusTypeId'],
+          properties: {
+            userIdentifier: {
+              type: 'string',
+              description: 'The user identifier for the user.'
+            },
+            identitySource: {
+              type: 'string',
+              description: 'The identity source for the user.'
+            },
+            requestId: {
+              type: 'number',
+              description: 'The id of the access request to update.'
+            },
+            requestStatusTypeId: {
+              type: 'number',
+              description: 'The status type id to set for the access request.'
+            },
+            roleIds: {
+              type: 'array',
+              items: {
+                type: 'number'
+              },
+              description:
+                'An array of role ids to add, if the access-request was approved. Ignored if the access-request was denied.'
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'Add system user roles to user OK.'
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+function updateAccessRequest(): RequestHandler {
+  return async (req, res) => {
+    defaultLog.debug({ label: 'updateAccessRequest', message: 'params', req_body: req.body });
+
+    const userIdentifier = req.body?.userIdentifier || null;
+    const identitySource = req.body?.identitySource || null;
+    const administrativeActivityId = Number(req.body?.requestId) || null;
+    const administrativeActivityStatusTypeId = Number(req.body?.requestStatusTypeId) || null;
+    const roleIds: number[] = req.body?.roleIds || [];
+
+    if (!userIdentifier) {
+      throw new HTTP400('Missing required body param: userIdentifier');
+    }
+
+    if (!identitySource) {
+      throw new HTTP400('Missing required body param: identitySource');
+    }
+
+    if (!administrativeActivityId) {
+      throw new HTTP400('Missing required body param: requestId');
+    }
+
+    if (!administrativeActivityStatusTypeId) {
```

Comment:
question:
do we not need to check for `!roleIds`?

Reply:
No, as this endpoint is generic to both approved and denied updates.  So in the case of a Denied request, we wouldn't pass any roles.